tests/stream.ts

```diff
@@ -74,6 +74,33 @@ describe('Stream', () => {
     });
   });
 
+  it('should be createable giving a custom producer function', (done) => {
+    const expected = [10, 20, 30];
+    let producerStopped: boolean = false;
+
+    const stream: Stream<number> = xs.create<number>((listener) => {
+      listener.next(10);
+      listener.next(20);
+      listener.next(30);
+      listener.complete();
+      return () => {
+        done();
```

Comment:
Yes this is just a proposal, and I showed that it can be implemented if needed. I think it brings not much complication, but allows to align with standard observable approach.